corehq/ex-submodules/casexml/apps/case/models.py

```diff
@@ -583,7 +585,13 @@ def _safe_replace_and_force_to_string(me, attr, val):
     
     def update_from_case_update(self, case_update, xformdoc, other_forms=None):
         other_forms = other_forms or {}
-        if case_update.has_referrals():
+        if is_deprecation(xformdoc):
+            # remove the form from actions and xform_ids (based on orig_id)
+            # and rebuild
+            self.xform_ids = [id for id in self.xform_ids if id != xformdoc.orig_id]
+            self.actions = [a for a in self.actions if a.xform_id != xformdoc.orig_id]
```

Comment:
it wasn't a completely explicit design decision so much as it looked a lot easier to implement. can add a comment if you think it's useful though.